File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.go


func topKFrequent(nums []int, k int) []int {

    frqmap := make(map[int]int)

    for _ , num := range nums {
        frqmap[num] ++
    }

    bucket := make([][]int , len(nums)+ 1)

    for k , v := range frqmap {
        bucket[v] = append(bucket[v] , k)
    }


    res := []int{}

    for i := len(bucket) - 1 ; i >= 0 && k > 0 ; i -- {
        for _ , num := range bucket[i] {
            res = append(res , num)
            k --
            if k == 0 {

                break
            }
        }
    }

    return res




}

File: Data Structures & Algorithms/concatenation-of-array/submission-1.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        n = len(nums)
        result = [0] * 2*n

        for i in range(2*n):
            if i < n:
                result[i] = nums[i]
            else:
                result[i] = nums[i-n]

        return result
